### PR TITLE
[495] associate trainee with provider in dttp

### PR DESCRIPTION
**Pull request by SamJamCul**
### Context
https://trello.com/c/S6jiBWaE/495-s-associate-trainee-with-provider-in-dttp

Eventually we'll need to ensure that the dttp_id on providers can't be Null, as they all need to correspond to a DTTP provider - for now though this isn't really doable

### Changes proposed in this pull request
* Add column to provider with dttp_id
* Add param to contact_params for dttp with provider dttp_id

### CODE REVIEW

**Comment by SamJamCul** [comment]
[[S] Associate trainee with provider in DTTP](https://trello.com/c/S6jiBWaE/495-s-associate-trainee-with-provider-in-dttp)

File: app/presenters/dttp/trainee_presenter.rb [hunk 1]
```diff
@@ -27,6 +27,7 @@ def contact_params
         "mobilephone" => trainee.phone_number,
         "dfe_EthnicityId@odata.bind" => "/dfe_ethnicities(#{dttp_ethnicity_id})",
         "dfe_DisibilityId@odata.bind" => "/dfe_disabilities(#{dttp_disability_id})",
+        "parentcustomerid_account@odata.bind" => "/accounts(#{trainee.provider.dttp_id})",
```
Comment (gpeng) [inline app/presenters/dttp/trainee_presenter.rb:30]: We could just do `provider.dttp_id` here as we're delegating missing to `trainee`. In fact, all of the `trainee.` are redundant here.

File: test.rb [hunk 2]
```diff
@@ -0,0 +1,7 @@
+require_relative 'config/environment'
```
Comment (gpeng) [inline test.rb:1]: 🤔

File: app/lib/dttp/code_sets/institutions.rb [hunk 3]
```diff
@@ -5,193 +5,193 @@ module CodeSets
...
+        "University of Northumbria at Newcastle" => { entity_id: "2a71f34a-2887-e711-80d8-005056ac45bb" },
+        "University of Nottingham" => { entity_id: "2c71f34a-2887-e711-80d8-005056ac45bb" },
+        "University of Plymouth" => { entity_id: "3071f34a-2887-e711-80d8-005056ac45bb" },
+        "University of South Wales" => { entity_id: "8723a753-7042-e811-80ff-3863bb3640b8" },
+        "University of Southampton" => { entity_id: "4f5b7f3b-7042-e811-80ff-3863bb3640b8" },
+        "University of Surrey" => { entity_id: "58228041-7042-e811-80ff-3863bb3640b8" },
+        "University of Sussex" => { entity_id: "3e71f34a-2887-e711-80d8-005056ac45bb" },
+        "University of Ulster" => { entity_id: "a823a753-7042-e811-80ff-3863bb3640b8" },
+        "University of Wales Trinity Saint David" => { entity_id: "b123a753-7042-e811-80ff-3863bb3640b8" },
+        "University of Wales, Cardiff" => { entity_id: "" }, # Not on the list
```
Comment (gpeng) [inline app/lib/dttp/code_sets/institutions.rb:182]: Have we got a plan for these missing ones?

**Review by gpeng — CHANGES_REQUESTED** [review]

File: test.rb (hunk 2 above)
Comment (jhenderson) [reply test.rb:1]: This is commited my mistake - will remove it.

File: app/lib/dttp/code_sets/institutions.rb (hunk 3 above)
Comment (jhenderson) [reply app/lib/dttp/code_sets/institutions.rb]: All entity ID's are populated now.

File: app/services/dttp/contact/create.rb [hunk 4]
```diff
@@ -20,6 +20,7 @@ def initialize(trainee:)
       def call
         response = Client.post("/contacts", body: trainee.contact_params.to_json)
         trainee.update!(dttp_id: OdataParser.entity_id(trainee.id, response))
+        trainee.dttp_id
```
Comment (defong) [inline app/services/dttp/contact/create.rb:23]: Odd...

Is there no test coverage?

Comment (jhenderson) [reply app/services/dttp/contact/create.rb]: We can actually remove this line now. We returned the `dttp_id` for manually testing. We're going to removing these classes anyway in favour of a batch request.

**Review by gpeng — APPROVED** [review]

**Review by defong — APPROVED** [review]

**Pull request merged**